packages/yarnpkg-parsers/tests/syml.test.js

```diff
@@ -21,13 +25,48 @@ describe(`Syml parser`, () => {
 
   it(`should merge duplicates`, () => {
     expect(
-      parseSyml(`    
+      parseSyml(`
       "lodash@npm:^4.17.20":
         version: 4.17.20
-      
+
       "lodash@npm:^4.17.20":
         version: 4.17.20
       `),
     ).toEqual({'lodash@npm:^4.17.20': {version: `4.17.20`}});
   });
 });
+
+describe('Yaml parse comparison (to be removed / reworked)', () => {
+  it(`should parse identically as js-yaml@3`, () => {
+
+    const lock = fs.readFileSync(`${__dirname}/../../../yarn.lock`, `utf8`);
+
+    // js-yaml@3 doesn't parse `key: true` as a boolean, but as a string.
+    // as far as I tested it makes the lock file invalid after install.
+    const reviver = (key, value) => {
+      if (value === true) {
+        return 'true';
+      } else if (value === false) {
+        return 'false';
+      }
+      return value;
+    }
+
+    const startJsYaml = performance.now();
+    const jsYamlParsed = parseWithJsYaml(lock, {schema: FAILSAFE_SCHEMA});
+    const endJsYaml = performance.now();
+    const jsYamlTime = endJsYaml - startJsYaml;
+
+    const startYaml = performance.now();
+    const yamlParsed = parseWithYaml(lock, reviver, {
+      uniqueKeys: false,
+      schema: 'failsafe',
+      customTags: []
+    })
+    const endYaml = performance.now();
+    const yamlTime = endYaml - startYaml;
+
+    expect(yamlParsed).toStrictEqual(jsYamlParsed);
+    expect(yamlTime).toBeLessThan(jsYamlTime);
```

Comment:
Current finding:

Expected:  < 64.09353400021791 ms (js-yaml)
Received:   513.0383080020547 ms (yaml)

Yes that might be a real blocker (at least for parsing yarn.lock which is not required). Adding both libs might not be a good move :(